paperscraper/lib.py

```diff
@@ -306,6 +319,96 @@ def parse_semantic_scholar_metadata(paper: dict[str, Any]) -> dict[str, Any]:
     }
 
 
+async def parse_google_scholar_metadata(
+    paper: dict[str, Any], session: ClientSession
+) -> dict[str, Any]:
+    """Parse raw paper metadata from google scholar into a more rich format."""
+    doi = paper["externalIds"].get("DOI", None)
+    key = None
+    if doi:
+        bibtex = await doi_to_bibtex(doi, session)
+        key = bibtex.split("{")[1].split(",")[0]
+        citation = format_bibtex(bibtex, key, clean=False)
+    if key is None:
+        # get citation by following link
+        # SLOW SLOW Using SerpAPI for this
+        async with session.get(
+            paper["inline_links"]["serpapi_cite_link"]
+            + f"&api_key={os.environ['SERPAPI_API_KEY']}"
+        ) as r:
```

Comment:
```suggestion
        async with session.get(
            paper["inline_links"]["serpapi_cite_link"],
            params={"api_key": os.environ["SERPAPI_API_KEY"]}
        ) as r:
```

Imo it's better to use URL params instead of `&`

Reply:
There are params in the link already - will they be correctly combined or will this insert an extra `&` - wasn't sure

Reply:
Yeah it's sort of like a slight of hand, it ends up using `&` behind the scenes, but it also takes care of things like URL encoding

Reply:
Feel free to ignore this comment